File: Luogu/P3717/P3717.cpp
//
// Created by hosibosi on 2021/3/25.
//

#include <iostream>
#include <vector>
#include <cmath>


int main(){
    int count = 0;
    int area, numDetector, radius;
    double distanceIn2P;
    std::cin >> area >> numDetector >> radius;
    std::vector<std::pair<int, int>> detectorPosition;
    for (auto i = 0; i < numDetector; ++i) {
        int dP_x, dP_y;
        std::cin >> dP_x >> dP_y;
        detectorPosition.emplace_back(std::make_pair(dP_x, dP_y));
    }
    for (auto position_x = 1; position_x <= area; ++position_x) {
        for (auto position_y = 1; position_y <= area; ++position_y) {
            for (auto dP : detectorPosition){
                distanceIn2P = sqrt(pow(position_x - dP.first, 2) + pow(position_y - dP.second, 2));
                if (distanceIn2P <= radius){
                    count++;
                    break;
                }
            }
        }

    }
    std::cout<< count << std::endl;
    return 0;
}

File: Luogu/P3717/P3717.py
from math import sqrt

area, num_detector, radius = map(int, input().split())
detector_list = []
count = 0
for i in range(num_detector):
    detector_list.append(tuple(map(int, input().split())))
for position_x in range(1, area + 1):
    for position_y in range(1, area + 1):
        for detector_position in detector_list:
            distance = sqrt(pow(detector_position[0] - position_x, 2) + pow(detector_position[1] - position_y, 2))
            if distance <= radius:
                count += 1
                break
print(count)